geopyspark/geopyspark_utils.py

```diff
@@ -47,6 +47,7 @@ def setup_environment():
     os.environ["PYSPARK_DRIVER_PYTHON"] = "python3"
     os.environ["PYSPARK_SUBMIT_ARGS"] = "--jars {} \
             --conf spark.ui.enabled=false \
+            --conf spark.serializer=org.apache.spark.serializer.KryoSerializer \
```

Comment:
Possibly of interest.  This is necessary because Python `lists` become `java.util.ArrayList`s which are evidently not serializable by the default mechanism.